### PR TITLE
Add min balance check for recipient - Closes #4848

### LINKED ISSUE #4848: Add MIN_REMAINING_BALANCE static property in the base transaction, and verify after updating the account
### Description

With the new dynamic fee system, `MIN_REMAINING_BALANCE ` should be adjustable.
It should add `MIN_REMAINING_BALANCE ` a static property in the base transaction, and after subtracting `balance` from the `sender`, it should check against this `MIN_REMAINING_BALANCE `

### Motivation

Motivation is described in https://github.com/LiskHQ/lips/blob/master/proposals/lip-0025.md

### Acceptance Criteria

- The base transaction should have this property set to 500000 Beddow in `bigint` as a static property
- When subtracting fee from sender account in the apply step, it should check if the remaining `balance` is equal or greater than specified `MIN_REMAINING_BALANCE `
- Test base transaction where sender balance is the same amount as minimum balance
- Test base transaction where sender balance is below the `MIN_REMAINING_BALANCE `
- Test base transaction where sender balance is above the `MIN_REMAINING_BALANCE `

- When recipient exists in transaction (only type 8), check resulting balance is is equal or greater than specified `MIN_REMAINING_BALANCE `
- Test transfer transaction where recipient resulting balance  is below the `MIN_REMAINING_BALANCE `

### Related issues
#4837

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #4848 

### How was it solved?

Add missing recipient balance check on the transfer transaction

### How was it tested?

New test case is covered in the test

### CODE REVIEW

File: elements/lisk-chain/test/unit/process.spec.ts [hunk 1]
```diff
@@ -650,7 +650,7 @@ describe('blocks/header', () => {
 				).rejects.toMatchObject([
 					expect.objectContaining({
 						message: expect.stringContaining(
-							'Account does not have enough LSK',
+							'Account does not have enough minimum remaining LSK',
```
Comment (pablitovicente) [inline elements/lisk-chain/test/unit/process.spec.ts]: Maybe show the value for minimum?

File: elements/lisk-transactions/test/base_transaction.spec.ts [hunk 2]
```diff
@@ -584,8 +584,8 @@ describe('Base transaction class', () => {
 			expect((errors as ReadonlyArray<TransactionError>)[0]).toBeInstanceOf(
 				TransactionError,
 			);
-			expect((errors as ReadonlyArray<TransactionError>)[0].message).toEqual(
-				`Account does not have enough LSK: ${defaultSenderAccount.address}, balance: 0`,
+			expect((errors as ReadonlyArray<TransactionError>)[0].message).toContain(
```
Comment (pablitovicente) [inline elements/lisk-transactions/test/base_transaction.spec.ts]: Maybe better to do exact match?

File: elements/lisk-chain/test/unit/process.spec.ts (hunk 1 above)
Comment (shuse2) [reply elements/lisk-chain/test/unit/process.spec.ts]: Updated the error message to include actual and expected

File: elements/lisk-transactions/test/base_transaction.spec.ts (hunk 2 above)
Comment (shuse2) [reply elements/lisk-transactions/test/base_transaction.spec.ts]: Checking the exact match of error message make the test too fragile IMO

**Pull request merged**